lib/manager/poetry/artifacts.js

```diff
@@ -0,0 +1,76 @@
+const upath = require('upath');
+const process = require('process');
+const fs = require('fs-extra');
+const { exec } = require('child-process-promise');
+
+module.exports = {
+  getArtifacts,
+};
+
+async function getArtifacts(
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config
+) {
+  await logger.debug(`poetry.getArtifacts(${packageFileName})`);
+  if (updatedDeps === undefined || updatedDeps.length < 1) {
+    logger.debug('No updated poetry deps - returning null');
+    return null;
+  }
+  const lockFileName = 'poetry.lock';
+  const existingLockFileContent = await platform.getFile(lockFileName);
+  if (!existingLockFileContent) {
+    logger.debug(`No ${lockFileName} found`);
+    return null;
+  }
+  const localPackageFileName = upath.join(config.localDir, packageFileName);
+  const localLockFileName = upath.join(config.localDir, lockFileName);
+  let stdout;
+  let stderr;
+  const startTime = process.hrtime();
+  try {
+    await fs.outputFile(localPackageFileName, newPackageFileContent);
+    logger.debug(`Updating ${lockFileName}`);
+    const cwd = config.localDir;
+    for (let i = 0; i < updatedDeps.length; i += 1) {
+      const dep = updatedDeps[i];
+      const cmd = `poetry update --lock --no-interaction ${dep}`;
```

Comment:
What does the documentation say this command does? e.g. it refreshes the lock entry for this particular dep but does not touch other dep locked versions?

Reply:
Yes, it only updates lock entry for `${dep}` in `poetry.lock` file and leaves other dependencies alone.
`--lock` option prevents it from installing packages.
`--no-interaction` makes it non-interactive, so it doesn't require any input on `stdin`.
See: https://poetry.eustace.io/docs/cli/#update